src/meta/src/manager/cluster.rs

```diff
@@ -176,6 +176,16 @@ where
             .generate::<{ IdCategory::Worker }>()
             .await? as WorkerId;
 
+        let transactional_id = match (core.available_transactional_ids.front(), r#type) {
+            (None, _) => {
+                return Err(MetaError::unavailable(
+                    "no available reusable machine id".to_string(),
+                ))
+            }
+            (Some(id), WorkerType::ComputeNode) => Some(*id),
+            _ => None,
+        };
```

Comment:
We only generate `transactional_id` for compute nodes, but it is highly possible that frontend nodes would use it to generate some id too. What about generating for all node types?

Reply:
Added the frontend.